pkg/objstore/gcs/gcs.go

```diff
@@ -32,29 +37,55 @@ const (
 // DirDelim is the delimiter used to model a directory structure in an object store bucket.
 const DirDelim = "/"
 
+// gcsConfig stores the configuration for gcs bucket.
+type gcsConfig struct {
+	Bucket string `yaml:"bucket"`
+}
+
 // Bucket implements the store.Bucket and shipper.Bucket interfaces against GCS.
 type Bucket struct {
+	logger   log.Logger
 	bkt      *storage.BucketHandle
 	opsTotal *prometheus.CounterVec
+	bucket   string
 
 	closer io.Closer
 }
 
 // NewBucket returns a new Bucket against the given bucket handle.
-func NewBucket(name string, cl *storage.Client, reg prometheus.Registerer) *Bucket {
+func NewBucket(logger log.Logger, ctx context.Context, conf []byte, reg prometheus.Registerer, component string) (*Bucket, error) {
```

Comment:
i think its convention to have `ctx` as the first variable in the method. See conclusion here https://blog.golang.org/context